CMakeLists.txt

```diff
@@ -639,43 +671,74 @@ install (
    NIH dep: lz4
 #]===================================================================]
 
-add_library (lz4 STATIC
-  src/lz4/lib/lz4.c
-  src/lz4/lib/lz4hc.c
-  src/lz4/lib/lz4frame.c
-  src/lz4/lib/xxhash.c)
-target_compile_definitions (lz4
-  PRIVATE XXH_NAMESPACE=LZ4_)
-add_library (NIH::lz4 ALIAS lz4)
-target_link_libraries (ripple_libs INTERFACE NIH::lz4)
-## pseudo-install our files so that dependent builds can find them
-file (MAKE_DIRECTORY
-    ${CMAKE_BINARY_DIR}/lz4/include
-    ${CMAKE_BINARY_DIR}/lz4/lib)
-target_include_directories (lz4 PUBLIC ${CMAKE_BINARY_DIR}/lz4/include)
-add_custom_command (TARGET lz4 POST_BUILD
-  COMMAND ${CMAKE_COMMAND} -E copy
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/lz4/lib/lz4.h
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/lz4/lib/lz4frame.h
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/lz4/lib/lz4hc.h
-    ${CMAKE_BINARY_DIR}/lz4/include
-  COMMAND ${CMAKE_COMMAND} -E copy
-    $<TARGET_FILE:lz4> ${CMAKE_BINARY_DIR}/lz4/lib)
+ExternalProject_Add (lz4
+  PREFIX ${nih_cache_path}
+  GIT_REPOSITORY https://github.com/lz4/lz4.git
+  GIT_TAG v1.8.2
+  SOURCE_SUBDIR contrib/cmake_unofficial
+  CMAKE_ARGS
+    -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+    -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+    $<$<BOOL:${CMAKE_VERBOSE_MAKEFILE}>:-DCMAKE_VERBOSE_MAKEFILE=ON>
+    -DCMAKE_DEBUG_POSTFIX=_d
+    $<$<NOT:$<BOOL:${is_multiconfig}>>:-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}>
+    -DBUILD_STATIC_LIBS=ON
+    -DBUILD_SHARED_LIBS=OFF
+    $<$<BOOL:${MSVC}>:
+      "-DCMAKE_C_FLAGS=-GR -Gd -fp:precise -FS -MP"
+      "-DCMAKE_C_FLAGS_DEBUG=-MTd"
+      "-DCMAKE_C_FLAGS_RELEASE=-MT"
+    >
+  LOG_BUILD ON
+  LOG_CONFIGURE ON
+  BUILD_COMMAND
+    ${CMAKE_COMMAND}
+    --build .
+    --config $<CONFIG>
+    --target lz4_static
+    $<$<VERSION_GREATER_EQUAL:${CMAKE_VERSION},3.12>:--parallel$<$<BOOL:${is_xcode}>: ${num_procs}>>
```

Comment:
What's the reason for only specifying `num_procs` for xcode?

Reply:
cmake 3.12 supports the `--parallel` option and it then passes the proper flag based on the generator being used. The number of jobs argument is optional as far as cmake is concerned and most generators are fine with that - but `xcodebuild` is not: it requires an integer argument or else it fails. If we are okay with the way `ProcessorCount` detects the num of cores, then I could just pass it always instead of letting the underlying build tool figure it out. This only impacts the EPs which hopefully aren't actually rebuilding every time.